fusilli-plugin/src/fusilli_plugin.cpp

```diff
@@ -0,0 +1,407 @@
+// Copyright 2025 Advanced Micro Devices, Inc.
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+//===----------------------------------------------------------------------===//
+//
+// This file is the main entry point for fusilli-plugin, implementations for all
+// required hipDNN engine plugin API functions live here.
+//
+//===----------------------------------------------------------------------===//
+
+// hipDNN logging expects COMPONENT_NAME to be defined
+#define COMPONENT_NAME FUSILLI_PLUGIN_NAME
+
+#include <flatbuffers/flatbuffers.h>
+#include <fusilli.h>
+#include <hip/hip_runtime.h>
+#include <hipdnn_sdk/data_objects/engine_details_generated.h>
+#include <hipdnn_sdk/data_objects/graph_generated.h>
+#include <hipdnn_sdk/data_objects/tensor_attributes_generated.h>
+#include <hipdnn_sdk/logging/Logger.hpp>
+#include <hipdnn_sdk/plugin/PluginApi.h>
+#include <hipdnn_sdk/plugin/PluginApiDataTypes.h>
+#include <hipdnn_sdk/plugin/PluginFlatbufferTypeHelpers.hpp>
+#include <hipdnn_sdk/plugin/PluginHelpers.hpp>
+#include <hipdnn_sdk/plugin/flatbuffer_utilities/EngineConfigWrapper.hpp>
+#include <hipdnn_sdk/plugin/flatbuffer_utilities/GraphWrapper.hpp>
+
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+#include <memory>
+#include <utility>
+
+#include "hipdnn_engine_plugin_execution_context.h"
+#include "hipdnn_engine_plugin_handle.h"
+#include "utils.h"
+
+using namespace hipdnn_plugin;
+
+// TODO(#2317): ensure single source of truth for plugin version
+static const char *fusilliPuluginVersion = "0.0.1";
+
+// s_lastError is thread_local static so can't be initialized in the header file
+// as the header file is included in many context. Clear the string here.
+thread_local char
+    PluginLastErrorManager::s_lastError[HIPDNN_PLUGIN_ERROR_STRING_MAX_LENGTH] =
+        "";
+
+extern "C" {
+
+// ----------------------------------------------------------------------
+// Implementations for the basic plugin API defined in
+// hipDNN/sdk/include/hipdnn_sdk/plugin/PluginApi.h
+// ----------------------------------------------------------------------
+
+hipdnnPluginStatus_t hipdnnPluginGetName(const char **name) {
+  LOG_API_ENTRY("name_ptr={:p}", static_cast<void *>(name));
+  FUSILLI_PLUGIN_CHECK_NULL(name);
+
+  *name = FUSILLI_PLUGIN_NAME;
+
+  LOG_API_SUCCESS_AUTO("pluginName={}", *name);
+  return HIPDNN_PLUGIN_STATUS_SUCCESS;
+}
```

Comment:
`hipDNN` uses a try catch pattern 
```C++
    return hipdnn_plugin::tryCatch([&, apiName = __func__]() {
        throwIfNull(name);

        *name = pluginName;

        LOG_API_SUCCESS(apiName, "pluginName={:p}", static_cast<void*>(name));
    });
```
fusilli (aka sharkfuser), and fusilli-plugin are using the LLVM style guide which discourages (well technically outright bans) exceptions, so I went with this slightly less exceptional (🤣) pattern.

Reply:
SGTM